cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/plugin/PluginService.java

```diff
@@ -213,15 +275,4 @@ public Object invoke(Object request) throws IOException, ClassNotFoundException,
     // cannot find the endpoint in plugin method. should not happen as this is checked earlier.
     throw new NotFoundException("Could not find the plugin method with the requested method endpoint {}", endpointName);
   }
-
-  @Override
-  public void close() throws IOException {
-    if (pluginInstantiator != null) {
-      Closeables.closeQuietly(pluginInstantiator);
-    }
-    if (parentClassLoader != null) {
-      Closeables.closeQuietly(parentClassLoader);
-    }
-    DirUtils.deleteDirectoryContents(stageDir);
```

Comment:
this means the directory never gets cleaned up.  The service should probably be a proper Service, where it creates the directory on startup and cleans it up on shutdown instead of creating the directory in the constructor.